src/hotspot/os/bsd/os_bsd.cpp

```diff
@@ -4024,6 +4006,61 @@ jlong os::thread_cpu_time(Thread *thread, bool user_sys_cpu_time) {
     return ((jlong)tinfo.user_time.seconds * 1000000000) + ((jlong)tinfo.user_time.microseconds * (jlong)1000);
   }
 #else
+#ifdef RUSAGE_THREAD
+  if (thread == Thread::current()) {
```

Comment:
I'm wondering how much value having this special case provides.  Is the thread being used often the current thread or does this execute rarely?  Can this thread be preempted between here and the call to getrusage?

Reply:
I'm assuming that if a thread queries its own cpu usage it would hit the RUSAGE_THREAD implementation. I kept it for OpenBSD since the getrusage implementation is a single syscall vs two syscalls and malloc/free. I have no strong feelings on keeping it or just leaving it for the clockgettime implementation (where it adds a bit of functionality). Probably an unnecessary micro optimization. If you have a preference let me know.